base/errorshow.jl

```diff
@@ -164,7 +164,7 @@ function showerror(io::IO, ex::MethodError)
     ft = typeof(f)
     name = ft.name.mt.name
     f_is_function = false
-    kwargs = ()
+    kwargs = NamedTuple()
```

Comment:
Also need to change the line below that was changed by #25658.

Reply:
Done. Does it look right now?

Reply:
This file in the revert is wrong. We want `kwargs` to be a non-specialized container so that we don't compile a new copy of `show_method_candidates`.

Reply:
Wasn't it already that way though? `IndexValue` is also specialized.

Reply:
I'm not saying I necessarily managed to catch all of this regression before. In v0.6, we copied it to a Vector{Any}. That's not strictly necessary though, as long we don't pass this as an argument to something that'll cause code-generation-specialization of our error-printing code.